src/places/place.ts

```diff
@@ -172,15 +289,117 @@ export class MigrationPlace {
         .then((response) => {
           const googlePlaces = [];
 
           const results = response.Results;
           if (results.length !== 0) {
             results.forEach(function (place) {
-              const newPlace = convertAmazonPlaceToGoogleNewPlace(place, fields);
+              const newPlace = convertAmazonPlaceToGoogleNewPlaceV1(place, fields);
+
+              // Set the requested language/region on the new Place if they had been specified in this search
+              if (language) {
+                newPlace.requestedLanguage = language;
+              }
+
+              googlePlaces.push(newPlace);
+            });
+          }
+
+          resolve({
+            places: googlePlaces,
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+
+          reject({
+            status: PlacesServiceStatus.UNKNOWN_ERROR,
+          });
+        });
+    });
+  }
+
+  // https://developers-dot-devsite-v2-prod.appspot.com/maps/documentation/javascript/reference/place#Place.searchNearby
+  public static searchNearby(
+    request: google.maps.places.SearchNearbyRequest,
+  ): Promise<{ places: google.maps.places.Place[] }> {
+    const locationRestriction = request.locationRestriction; // required
+    const fields = request.fields; // required for searchNearby, but not other requests
+    const language = request.language; // optional
+    const region = request.region; // optional
+    const maxResultCount = request.maxResultCount; // optional
+    const includedTypes = request.includedTypes; // optional
+    const excludedTypes = request.excludedTypes; // optional
+
+    // locationRestriction is google.maps.Circle | google.maps.CircleLiteral
+    const circle = new MigrationCircle(locationRestriction);
+    const center = circle.getCenter();
+    const centerLngLat = LatLngToLngLat(center);
+    const radius = circle.getRadius();
+
+    const input: SearchNearbyRequest = {
+      QueryPosition: centerLngLat,
+      QueryRadius: radius,
+      AdditionalFeatures: ["Contact", "TimeZone"], // "Contact" is needed for contact details and opening hours, "TimeZone" is needed for place's time zone
+    };
+
+    if (language) {
+      input.Language = language;
+    }
+
+    if (maxResultCount) {
+      input.MaxResults = maxResultCount;
+    }
+
+    if (includedTypes) {
+      const categories = includedTypes.flatMap((type) => {
+        return getAllAmazonPlaceTypesFromGoogle(type);
+      });
+
+      input.Filter = {
+        IncludeCategories: categories,
+      };
+    }
+
+    if (excludedTypes) {
+      const categories = excludedTypes.flatMap((type) => {
+        return getAllAmazonPlaceTypesFromGoogle(type);
+      });
+
+      input.Filter = {
+        ExcludeCategories: categories,
+      };
+    }
+
+    return new Promise((resolve, reject) => {
+      const command = new SearchNearbyCommand(input);
+
+      MigrationPlace._client
+        .send(command)
+        .then((response) => {
```

Comment:
Do we specifically want these as non-blocking async promises? If not, then for blocking the await syntax is usually more readable compared to then chains.

Reply:
Yes, Google's APIs of these are non-blocking so our counterparts should be non-blocking as well.